impeller/renderer/backend/vulkan/command_encoder_vk.h

```diff
@@ -36,15 +38,19 @@ class CommandEncoderVK {
 
   void PopDebugGroup() const;
 
+  std::optional<vk::DescriptorSet> AllocateDescriptorSet(
+      const vk::DescriptorSetLayout& layout);
+
  private:
   friend class ContextVK;
 
   vk::Device device_ = {};
   vk::Queue queue_ = {};
   vk::UniqueCommandBuffer command_buffer_;
-  std::vector<std::shared_ptr<SharedObjectVK>> tracked_objects_;
-  std::vector<std::shared_ptr<const DeviceBuffer>> tracked_buffers_;
-  std::vector<std::shared_ptr<const Texture>> tracked_textures_;
+  DescriptorPoolVK desc_pool_;
```

Comment:
I might be misreading this, but won't this pool rapidly leak because `CommandEncoderVK` has the same long lifetime as ContextVK? Should this expanding pool have a shorter lifetime (like per-frame or per-renderpass)?

Reply:
NVM, I was misreading this. New `CommandEncoderVK` are actually created per-command buffer.

Reply:
I was modeling this encoder on MTLRenderCommandEncoder.